internal/role/roles.go

```diff
@@ -0,0 +1,64 @@
+package role
+
+import (
+	"strings"
+)
+
+type Role string
+
+const (
+	Shell    Role = "shell"
+	Actions  Role = "actions"
+	Download Role = "download"
+)
+
+var AllRoles = []Role{Shell, Actions, Download}
+
+type UserRoles []Role
+
+func (ur UserRoles) Exists() bool {
+	return len(ur) > 0
+}
+
+func IsValidRole(role Role) bool {
+	for _, r := range AllRoles {
+		if role == r {
+			return true
+		}
+	}
+	return false
+}
+
+func ParseUserRole(commaValues string) UserRoles {
+	if commaValues == "" {
+		return UserRoles{}
+	}
+
+	roles := make(UserRoles, 0)
+	seen := make(map[Role]struct{})
+
+	for r := range strings.SplitSeq(commaValues, ",") {
+		role := Role(strings.TrimSpace(r))
+		if role == "" {
+			continue
+		}
+		if !IsValidRole(role) {
+			continue
+		}
+
+		if _, exists := seen[role]; !exists {
+			seen[role] = struct{}{}
+			roles = append(roles, role)
+		}
+	}
+	return roles
+}
+
+func (ur UserRoles) HasRole(role Role) bool {
```

Comment:
If I am understanding this correctly, it is not possible to have no roles. Right? I wonder if none should be an option.

Reply:
If there are no roles, global parameters will be used